apps/gimbtonomy/main.cpp

```diff
@@ -47,33 +80,87 @@ void handleSetNeoPixelColor(CANMsg *p_newMsg) {
   }
 }
 
-void processCANMsg(CANMsg *p_newMsg) {
-  switch (p_newMsg->id) {
-    case CANID::SET_NEOPIXEL:
-      // Send an acknowledgement CANMsg back to the Jetson
-      printf("Sending acknowledgement message\r\n");
-      txMsg.clear();
-      txMsg.id = CANID::NEOPIXEL_ACK;
-      txMsg << true;
-      can.write(txMsg);
-      printf("Updating neo pixels\r\n");
-      handleSetNeoPixelColor(p_newMsg);
-      break;
-    default:
-      printf("Recieved unimplemented command for neopixels\r\n");
-      break;
+// Incoming message processor
+void rxCANProcessor() {
+  const int rxPeriod_millisec = 2;
+
+  float pan_pos = 0.0, pan_speed = 0.0, tilt_pos = 0.0, roll_pos = 0.0;
+
+  while (true) {
+    if (can1.read(rxMsg)) {
+      switch (rxMsg.id) {
+        case CANID::SET_GIMBAL_PAN_POS:
```

Comment:
CAN commands to set the control mode of this actuator should also be exposed. EX to allow direct control incase of encoder failure.